lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -0,0 +1,293 @@
+        monkeypatch.setattr(
+            "pbench.server.unpack_tarballs.get_tarball_md5", lambda tb: "my_md5_string"
+        )
+        monkeypatch.setattr(FileTree, "unpack", mock_unpack)
+        obj = UnpackTarballs(server_config, make_logger)
+        result = obj.unpack_tarballs(self.min_size, self.max_size)
+        for tar in self.tarlist:
+            assert (
+                f"Unpacking of tarball {tar} failed: An error occurred while unpacking tarball: tar exited with status 1"
+                in caplog.text
+            )
+        assert result.total == 2
+        assert result.success == 0
+
+    def test_unpack_tarballs_unlink_failure(
+        self, monkeypatch, server_config, make_logger, caplog
+    ):
+        """Show that when unlinking failure leads to FileNotFoundError,
+        it is handled successfully."""
+
+        def mock_unlink(path):
+            """
+            Mock of os.unlink() function
+
+            Args:
+                path: tarball path in 'TO-INDEX' state directory
+
+            Returns:
+                Raises FileNotFoundError Exception
+            """
+            raise FileNotFoundError(f"No such file or directory: '{path}'")
+
+        monkeypatch.setattr(Path, "glob", lambda path, args: self.tarlist)
+        monkeypatch.setattr(Path, "resolve", lambda self, strict: self)
+        monkeypatch.setattr(Path, "stat", self.mock_stat)
+        monkeypatch.setattr(
+            "pbench.server.unpack_tarballs.get_tarball_md5", lambda tb: "my_md5_string"
+        )
+        monkeypatch.setattr(FileTree, "unpack", lambda self, dataset_id: None)
+        monkeypatch.setattr(os, "symlink", lambda path, dest: None)
+        monkeypatch.setattr(os, "unlink", mock_unlink)
+        obj = UnpackTarballs(server_config, make_logger)
+        result = obj.unpack_tarballs(self.min_size, self.max_size)
+        for tar in self.tarlist:
+            assert (
+                f"Error in removing symlink from TO-UNPACK state. No such file or directory: '{server_config.ARCHIVE / 'ABC' / obj.LINKSRC / Path(tar).name}'"
+                in caplog.text
+            )
+        assert result.total == 2
+        assert result.success == 0
+
+    def test_unpack_tarballs_success(
+        self, monkeypatch, server_config, make_logger, caplog
+    ):
+        """Show that when unpacking goes smoothly we can count the
+        successfully unpacked tarballs."""
+
+        monkeypatch.setattr(Path, "glob", lambda path, args: self.tarlist)
+        monkeypatch.setattr(Path, "resolve", lambda self, strict: self)
+        monkeypatch.setattr(Path, "stat", self.mock_stat)
+        monkeypatch.setattr(
+            "pbench.server.unpack_tarballs.get_tarball_md5", lambda tb: "my_md5_string"
+        )
+        monkeypatch.setattr(FileTree, "unpack", lambda self, dataset_id: None)
+        monkeypatch.setattr(os, "symlink", lambda path, dest: None)
+        monkeypatch.setattr(os, "unlink", lambda path: None)
```

Comment:
Rather than mocking `get_tarball_md5()`, I recommend mocking `FileTree.__init__()`.

Likewise, rather than mocking `os.symlink()` and `os.unlink()`, I recommend mocking `Tarball.update_symlink()`.